### PR TITLE
Feature/preceding following signs

### PR DESCRIPTION
**Pull request by Vas9ka**
* Added endpoint to get preceding or following signs for any sign and sort key 
E.g., "/signs/BAR/following/NeoAssyrianOffset"

### CODE REVIEW

File: ebl/signs/infrastructure/mongo_sign_repository.py [hunk 1]
```diff
@@ -101,6 +107,10 @@ def make_sign(self, data, **kwargs) -> Sign:
         return Sign(**data)
 
 
+class OrderedSignsSchema(Schema):
```
Comment (fsimonjetz) [inline ebl/signs/infrastructure/mongo_sign_repository.py:110]: `OrderedSignsSchema` no longer needed.

File: ebl/signs/infrastructure/mongo_sign_repository.py [hunk 2]
```diff
@@ -123,6 +133,74 @@ def find(self, name: SignName) -> Sign:
...
+                    "$match": {
+                        f"sortKeys.{sort_era}": {
+                            "$gte": range_start,
+                            "$lte": range_end,
+                        }
+                    }
+                },
+                {
+                    "$project": {
+                        "sign": 1,
```
Comment (fsimonjetz) [inline ebl/signs/infrastructure/mongo_sign_repository.py:166]: Looks like `sign` and `lists` are not used, if so they can be removed from the projection.

File: ebl/signs/infrastructure/mongo_sign_repository.py [hunk 3]
```diff
@@ -123,6 +133,74 @@ def find(self, name: SignName) -> Sign:
...
+                                        }
+                                    }
+                                },
+                                "in": {"$arrayElemAt": ["$$filteredResult.number", 0]},
+                            }
+                        },
+                    }
+                },
+                {"$sort": {"sort_key": 1}},
+                {"$group": {"_id": "1", "signs": {"$push": "$$ROOT"}}},
```
Comment (fsimonjetz) [inline ebl/signs/infrastructure/mongo_sign_repository.py:188]: I guess the purpose of `"_id": "1"` is to aggregate _all_ values? If so, the canonical way is to use `"_id": None` (at least that's how I did it in other cases).

File: ebl/tests/signs/test_sign_repository.py [hunk 4]
```diff
@@ -153,7 +153,7 @@ def sign_si_2(mongo_sign_si_2):
     return SignSchema(unknown=EXCLUDE).load(mongo_sign_si_2)
 
 
-def test_create(database, sign_repository, sign_igi):
+def teste(database, sign_repository, sign_igi):
```
Comment (fsimonjetz) [inline ebl/tests/signs/test_sign_repository.py:156]: typo?

Comment (Vas9ka) [reply ebl/tests/signs/test_sign_repository.py:156]: Yep, also noticed rn

File: ebl/signs/infrastructure/mongo_sign_repository.py (hunk 1 above)
Comment (Vas9ka) [reply ebl/signs/infrastructure/mongo_sign_repository.py:110]: Yes, forgot to exclude

File: ebl/signs/infrastructure/mongo_sign_repository.py (hunk 2 above)
Comment (Vas9ka) [reply ebl/signs/infrastructure/mongo_sign_repository.py:166]: Will do

File: ebl/signs/infrastructure/mongo_sign_repository.py (hunk 3 above)
Comment (Vas9ka) [reply ebl/signs/infrastructure/mongo_sign_repository.py:188]: Ok

File: ebl/signs/infrastructure/mongo_sign_repository.py [hunk 5]
```diff
@@ -123,6 +133,74 @@ def find(self, name: SignName) -> Sign:
...
+                                "vars": {
+                                    "filteredResult": {
+                                        "$filter": {
+                                            "input": "$lists",
+                                            "as": "item",
+                                            "cond": {"$eq": ["$$item.name", "MZL"]},
+                                        }
+                                    }
+                                },
+                                "in": {"$arrayElemAt": ["$$filteredResult.number", 0]},
```
Comment (fsimonjetz) [inline ebl/signs/infrastructure/mongo_sign_repository.py:182]: This is totally fine, and there's also the shorthand [`$first`](https://www.mongodb.com/docs/v5.0/reference/operator/aggregation/first-array-element/).

**Review by fsimonjetz — APPROVED** [review]
LGTM

**Pull request merged**